pkg/blueprint/filesystem_customizations.go

```diff
@@ -10,49 +10,43 @@ import (
 )
 
 type FilesystemCustomization struct {
-	Mountpoint string `json:"mountpoint,omitempty" toml:"mountpoint,omitempty"`
-	MinSize    uint64 `json:"minsize,omitempty" toml:"minsize,omitempty"`
+	Mountpoint string
+	MinSize    uint64
 }
 
-func (fsc *FilesystemCustomization) UnmarshalTOML(data interface{}) error {
-	d, ok := data.(map[string]interface{})
-	if !ok {
-		return fmt.Errorf("customizations.filesystem is not an object")
-	}
+type filesystemCustomizationMarshaling struct {
+	Mountpoint string         `json:"mountpoint,omitempty" toml:"mountpoint,omitempty"`
+	MinSize    datasizes.Size `json:"minsize,omitempty" toml:"minsize,omitempty"`
+}
 
-	switch d["mountpoint"].(type) {
-	case string:
-		fsc.Mountpoint = d["mountpoint"].(string)
-	default:
-		return fmt.Errorf("TOML unmarshal: mountpoint must be string, got \"%v\" of type %T", d["mountpoint"], d["mountpoint"])
-	}
-	minSize, err := decodeSize(d["minsize"])
+func (fsc *FilesystemCustomization) UnmarshalTOML(data any) error {
+	// This is the most efficient way to reuse code when unmarshaling
+	// structs in toml, it leaks json errors which is a bit sad but
+	// because the toml unmarshaler gives us not "[]byte" but an
+	// already pre-processed "any" we cannot just unmarshal into our
+	// "fooMarshaling" struct and reuse the result so we resort to
+	// this hack^Wworkaround (but toml will go long term anyway).
```

Comment:
Let's not be too silly in comments that might stick around for years or be read by people who might not fully get the meaning of `^W` for corrections and just call it a "workaround" (or a "hack", if you prefer).

Also, the phrase in parentheses should probably be "but toml will go *away* long term anyway", right?

Reply:
Thanks, fixed and force pushed together with a fix for the https://github.com/osbuild/images/pull/1055